### DISCUSSION TITLE
UQL ISSUES USING XML

### DISCUSSION THREAD

**Discussion by gdiana** [grafana/grafana-infinity-datasource#289] [Q&A]
Hi,

I thought I would play around with the UQL using XML and am having run into an odd issue.

I am using Grafana and UQL on the MAC and PC (win10 and Server 2012) and find that UQL works great on the MAC but using the identical queries on the PC I just get a query error.

I am using the latest Grafana version and Infinity Datasource and am experiencing no others problems.

What is odd is why it works on the MAC and not PC (win 10 and Win Server 2012)which seems to suggest some OS dependency.

Here is a screen capture that shows the xml query data being parsed

![image](https://user-images.githubusercontent.com/13445152/159924012-6aa8f584-9c48-4b31-b39f-f1c95d8e4634.png)

But as soon as I enter scope to access the table data I get and there is no data in the inspector either

![image](https://user-images.githubusercontent.com/13445152/159924341-85fa26b9-d248-4fc3-9e2f-b22be4a0d76b.png)

However, the query does parse back the query data as shown below

![image](https://user-images.githubusercontent.com/13445152/159924888-175e38c9-e396-4597-83a7-4a8079ba5c6f.png)

Weird but true!

Cheers

Greg Diana

**Comment by yesoreyeram** [comment]
@gdiana - Do you mind posting a sample XML for me to test? Grafana running on windows or browser on windows?

**Reply by gdiana** [reply]
@yesoreyeram 

GRAFANA is installed and run on my remote server which has Server 2012 and works fine.

I tried running Grafana on my mac Laptop using Google Chrome to connect to the remote server and UQL works but on my Windows 10 PC at home using Google Browser to connect to the remote server UQL plays up and the same on my remote Windows Server 2012 using remote desktop and Google Browser.

root is **xml.meters.meter** in UQL   and in xml it is    **xml.meters[0].meter**


[XMLSAMPLE.txt](https://github.com/yesoreyeram/grafana-infinity-datasource/files/8344014/XMLSAMPLE.txt)

**Reply by gdiana** [reply]
**UPDATE**

I did all UQL mods on MAC and then saved the dashboard. When I ran the modified dashboard on the Laptop with Win 10 the dashboard loaded fine and UQL worked/s fine too

But what I have said previously still holds. It is strange and I am sure it has something to do with caching?? as I have noted elsewhere before when I first started using XML.

What I have seen does not make sense!!

**Reply by gdiana** [reply]
Ok so will wait for the \r as it will resolve this issue. However, editing ON MAC and saving works fine on Windows.

Thanks again!👍
